draft-gpew-priv-ppm.md

```diff
@@ -900,37 +900,70 @@ The named parameters are:
-(This request may too have been made in advance.)
-
-Next, the leader computes its own aggregate share by aggregating all of the
-verified output shares that fall within the batch interval. Finally, it responds
-with HTTP status 200 and a body consisting of a CollectResp message:
-
-[OPEN ISSUE: What happens if this all takes a really long time.]
+~~~
+struct {
+  Url collect_job;
+} CollectResp;
+~~~
 
-[TODO: Decide if and how the collector's request is authenticated.]
+`CollectResp.collect_job` is a URL identifying the collect job that can be
+polled by the collector. It SHOULD contain an unpredictable component to
+prevent enumeration attacks against a leader's collect results. The HTTP
+response MAY include a Retry-After header field to suggest a polling interval to
+the client.
+
+The leader then begins working with the helper to prepare the shares falling
+into `CollectReq.batch_interval` (or continues this process, depending on the
+VDAF) as described in {{pa-aggregate}}. After receiving a `CollectResp`, the
+collector periodically makes HTTP GET requests to `CollectResp.collect_job` to
+check on the status of the collect job and eventually obtain the result. If the
+collect job is not finished yet, the leader responds with HTTP status 202
+Accepted and a `CollectResp`. The `CollectResp` MAY contain a different
+`collect_job` than previous `CollectResp`s and the HTTP response MAY include a
+`Retry-After` header field to suggest a polling interval to the collector.
+
+Once all the necessary reports have been prepared, the leader obtains the
+helper's encrypted aggregate share for the batch interval by sending an
+`AggregateShareReq` to the helper as described in {{aggregate-share-request}}.
+The leader then computes its own aggregate share by aggregating all of the
+prepared output shares that fall within the batch interval.
+
+When both aggregator's shares are successfully obtained, then the leader
+responds to subsequent HTTP GET requests to `CollectResp.collect_job` with HTTP
+status 200 OK and a body consisting of a `CollectResult`:
 
 ~~~
 struct {
-  EncryptedAggregateShare shares<1..2^16-1>;
-} CollectResp;
+  EncryptedAggregateShare shares<1..2^24-1>;
+} CollectResult;
 ~~~
 
 * `shares` is a vector of `EncryptedAggregateShare`s, as described in
   {{aggregate-share-request}}, except that for the leader's share, the `task_id`
   and `batch_interval` used to encrypt the `AggregateShare` are obtained from
   the `CollectReq`.
 
+If obtaining aggregate shares fails, then the leader responds to subsequent HTTP
+GET requests to `CollectResp.collect_job` with an HTTP error status and a
+problem document as described in {{errors}}.
+
+The leader SHOULD retain aggregate shares for a reasonable amount of time. Even
+after serving a `CollectResult` to the collector, leaders SHOULD continue to
+respond to further GET requests to `CollectResp.collect_job` in case the
+collector does not receive the response the first time.
```

Comment:
I see why you are doing this and you have to for E2E reasons, but this seems hard to operationalize. I sugegest that instead we want a delete.

Reply:
I consulted ACME for inspiration here: [Section 7.4.2 Downloading the Certificate](https://datatracker.ietf.org/doc/html/rfc8555#section-7.4.2). ACME doesn't specify how long the certificate should be available for at all (and technically allows an ACME server to _never_ provide the cert). Maybe we can get away with saying nothing, too. In any case I wrote up the DELETE idea.